numpy/ma/tests/test_core.py

```diff
@@ -759,6 +759,16 @@ def test_oddfeatures_3(self):
         atest[idx] = btest[idx]
         assert_equal(atest, [20])
 
+    @pytest.mark.parametrize("dtype", [np.int8, np.uint8, np.complex64])
+    def test_filled_with_basic_dtype(self, dtype):
+        # Function may change dtype of array with
+        # some basic type, see issue 27269
+        # https://github.com/numpy/numpy/issues/27269
+
+        # Test is the filled array has the same type
+        a = np.arange(1, 7).reshape(2, 3).astype(type)
+        assert masked_where(a <= 3, a).filled().dtype == type
```

Comment:
```suggestion
        a = np.arange(1, 7).reshape(2, 3).astype(dtype)
        assert masked_where(a <= 3, a).filled().dtype == dtype
```
To be consistent with the argument name in parametrize above.
Also, `type` is a built-in function of python, so it is better not to use it as a variable name.

Reply:
Thank you for pointing that out—I made a silly mistake, and the tests coincidentally passed. I've corrected it in the latest PR.